app/src/main-process/squirrel-updater.ts

```diff
@@ -72,30 +80,52 @@ function getBinPath(): string {
  * rewrite the trampoline to point to the new, version-specific path. Bingo
  * bango Bob's your uncle.
  */
-async function writeCLITrampoline(): Promise<string> {
-  const binPath = getBinPath()
-  const appFolder = Path.resolve(process.execPath, '..')
-  const versionedPath = Path.relative(
+async function writeBatchScriptCLITrampoline(binPath: string): Promise<void> {
+  const versionedPath = resolveVersionedPath(
     binPath,
-    Path.join(appFolder, 'resources/app/static/github.bat')
+    'resources/app/static/github.bat'
   )
-  const trampline = `@echo off\n"%~dp0\\${versionedPath}" %*`
+
+  const trampoline = `@echo off\n"%~dp0\\${versionedPath}" %*`
   const trampolinePath = Path.join(binPath, 'github.bat')
-  return new Promise<string>((resolve, reject) => {
-    Fs.ensureDir(binPath, err => {
+
+  return new Promise<void>((resolve, reject) => {
+    Fs.writeFile(trampolinePath, trampoline, err => {
```

Comment:
(I should have raised this previously, sorry :hurtrealbad:)

I think we should make an async `writeFile` (or whatever) in `file-system`. I bet we're using this promise wrapping in a number of places at this point.

Reply:
All good. Yeah, it'll definitely get some reuse elsewhere.